presto-main/src/test/java/com/facebook/presto/operator/TestPositionLinks.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.RowPagesBuilder;
+import com.facebook.presto.spi.Page;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestPositionLinks
+{
+    private static final Page TEST_PAGE = getOnlyElement(RowPagesBuilder.rowPagesBuilder(BIGINT).addSequencePage(20, 0).build());
+
+    @Test
+    public void testPositionLinksList()
+    {
+        PositionLinks.Builder builder = PositionLinksList.builder(1000);
+
+        assertFalse(builder.link(1, 0));
+        assertFalse(builder.link(2, 1));
+        assertFalse(builder.link(3, 2));
+
+        assertFalse(builder.link(11, 10));
+        assertFalse(builder.link(12, 11));
+
+        PositionLinks positionLinks = builder.build();
+
+        assertEquals(positionLinks.start(3, 0, TEST_PAGE), 3);
+        assertEquals(positionLinks.next(3, 0, TEST_PAGE), 2);
+        assertEquals(positionLinks.next(2, 0, TEST_PAGE), 1);
+        assertEquals(positionLinks.next(1, 0, TEST_PAGE), 0);
+
+        assertEquals(positionLinks.start(4, 0, TEST_PAGE), 4);
+        assertEquals(positionLinks.next(4, 0, TEST_PAGE), -1);
+
+        assertEquals(positionLinks.start(12, 0, TEST_PAGE), 12);
+        assertEquals(positionLinks.next(12, 0, TEST_PAGE), 11);
+        assertEquals(positionLinks.next(11, 0, TEST_PAGE), 10);
+    }
+
+    @Test
+    public void testSortedPositionLinks()
+    {
+        SortedPositionLinks.Builder builder = SortedPositionLinks.builder(
+                1000,
```

Comment:
yes, but I don't like it. This seems more easier to understand for me and it matches to reversed comparator in test below.